[2025-03-06T15:35:07.190+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T15:35:07.217+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T15:35:02.987627+00:00 [queued]>
[2025-03-06T15:35:07.228+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T15:35:02.987627+00:00 [queued]>
[2025-03-06T15:35:07.231+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T15:35:07.244+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 15:35:02.987627+00:00
[2025-03-06T15:35:07.251+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T15:35:02.987627+00:00', '--job-id', '13', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmp4d8eefmr']
[2025-03-06T15:35:07.254+0000] {standard_task_runner.py:105} INFO - Job 13: Subtask load_exchangeRate_python_job
[2025-03-06T15:35:07.255+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=1029) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T15:35:07.256+0000] {standard_task_runner.py:72} INFO - Started process 1030 to run task
[2025-03-06T15:35:07.312+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T15:35:02.987627+00:00 [running]> on host e9dfa152b6ab
[2025-03-06T15:35:07.409+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T15:35:02.987627+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T15:35:02.987627+00:00'
[2025-03-06T15:35:07.410+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T15:35:07.438+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T15:35:07.439+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T15:35:10.491+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T15:35:10.495+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T15:35:10.495+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkContext: Java version 17.0.14
[2025-03-06T15:35:10.545+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T15:35:10.624+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceUtils: ==============================================================
[2025-03-06T15:35:10.625+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T15:35:10.626+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceUtils: ==============================================================
[2025-03-06T15:35:10.627+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T15:35:10.644+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T15:35:10.651+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T15:35:10.652+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T15:35:10.695+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T15:35:10.695+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T15:35:10.696+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T15:35:10.697+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T15:35:10.697+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T15:35:10.905+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO Utils: Successfully started service 'sparkDriver' on port 42561.
[2025-03-06T15:35:10.929+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T15:35:10.963+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T15:35:10.981+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T15:35:10.982+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T15:35:10.986+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:10 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T15:35:11.007+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-e3d93f8d-6e10-42b9-bc7e-4a4c781cd9dc
[2025-03-06T15:35:11.020+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T15:35:11.035+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T15:35:11.150+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T15:35:11.215+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T15:35:11.250+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 ERROR SparkContext: Failed to add file:///opt/bitnami/spark/jars/postgresql-42.7.5.jar to Spark environment
[2025-03-06T15:35:11.251+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.7.5.jar not found
[2025-03-06T15:35:11.251+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T15:35:11.252+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T15:35:11.252+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T15:35:11.253+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T15:35:11.254+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T15:35:11.254+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T15:35:11.255+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T15:35:11.256+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T15:35:11.256+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T15:35:11.257+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T15:35:11.258+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T15:35:11.258+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T15:35:11.259+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T15:35:11.259+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T15:35:11.260+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T15:35:11.261+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T15:35:11.261+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T15:35:11.262+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T15:35:11.262+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T15:35:11.263+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T15:35:11.263+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T15:35:11.264+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T15:35:11.323+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.4:7077...
[2025-03-06T15:35:11.361+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO TransportClientFactory: Successfully created connection to /172.18.0.4:7077 after 20 ms (0 ms spent in bootstraps)
[2025-03-06T15:35:11.436+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306153511-0001
[2025-03-06T15:35:11.440+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306153511-0001/0 on worker-20250306150809-172.18.0.7-35707 (172.18.0.7:35707) with 2 core(s)
[2025-03-06T15:35:11.443+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306153511-0001/0 on hostPort 172.18.0.7:35707 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T15:35:11.444+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306153511-0001/1 on worker-20250306150810-172.18.0.10-33947 (172.18.0.10:33947) with 2 core(s)
[2025-03-06T15:35:11.445+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306153511-0001/1 on hostPort 172.18.0.10:33947 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T15:35:11.448+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 41103.
[2025-03-06T15:35:11.449+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO NettyBlockTransferService: Server created on e9dfa152b6ab:41103
[2025-03-06T15:35:11.450+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T15:35:11.456+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, e9dfa152b6ab, 41103, None)
[2025-03-06T15:35:11.460+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO BlockManagerMasterEndpoint: Registering block manager e9dfa152b6ab:41103 with 434.4 MiB RAM, BlockManagerId(driver, e9dfa152b6ab, 41103, None)
[2025-03-06T15:35:11.463+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, e9dfa152b6ab, 41103, None)
[2025-03-06T15:35:11.465+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, e9dfa152b6ab, 41103, None)
[2025-03-06T15:35:11.488+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306153511-0001/0 is now RUNNING
[2025-03-06T15:35:11.490+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306153511-0001/1 is now RUNNING
[2025-03-06T15:35:11.776+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:11 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T15:35:12.139+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:12 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T15:35:12.150+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:12 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T15:35:13.727+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T15:35:13.728+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 65, in <module>
[2025-03-06T15:35:13.729+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T15:35:13.730+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T15:35:13.731+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T15:35:13.732+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T15:35:13.738+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T15:35:13.754+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T15:35:13.866+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T15:35:13.867+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T15:35:13.887+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO SparkUI: Stopped Spark web UI at http://e9dfa152b6ab:4040
[2025-03-06T15:35:13.894+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T15:35:13.900+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T15:35:13.923+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T15:35:13.935+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO MemoryStore: MemoryStore cleared
[2025-03-06T15:35:13.936+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO BlockManager: BlockManager stopped
[2025-03-06T15:35:13.943+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T15:35:13.947+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T15:35:13.956+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T15:35:13.957+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T15:35:13.957+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-a3aa5f23-f5b9-431b-baa2-6ae0d73e1add/pyspark-02fa24ab-2628-4adb-a882-f7e39c9cf508
[2025-03-06T15:35:13.961+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-a3aa5f23-f5b9-431b-baa2-6ae0d73e1add
[2025-03-06T15:35:13.965+0000] {spark_submit.py:641} INFO - 25/03/06 15:35:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-3283a302-f5d5-4d7d-84c2-c13601ec9cd8
[2025-03-06T15:35:14.002+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T15:35:14.006+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T15:35:02.987627+00:00, execution_date=20250306T153502, start_date=20250306T153507, end_date=20250306T153514
[2025-03-06T15:35:14.039+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T15:35:14.040+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 13 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 1030)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.4:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T15:35:14.087+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T15:35:14.112+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T15:35:14.148+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
